common/flatpak-dir.c

```diff
@@ -2694,6 +2696,38 @@ get_common_pull_options (GVariantBuilder     *builder,
                          g_variant_new_variant (g_variant_new_uint32 (update_freq)));
 }
 
+static void
+flatpak_dir_cache_clean (OstreeRepo *repo)
+{
+  int dfd;
+  g_autoptr (GFile) file = NULL;
+
+  dfd = ostree_repo_get_dfd (repo);
+  file = g_file_new_for_path (glnx_fdrel_abspath (dfd, "tmp"));
+
+  flatpak_rm_rf (file, NULL, NULL);
+}
```

Comment:
I'm thinking, in the pull case, we should only remove *our* staging directory, not everything.

In a generic, we're low on disk case we could delete everything, but we should take the repo lock in exclusive mode to avoid running parallel operations mysteriously failing.